src/main/java/gyro/aws/ec2/TransitGatewayMulticastDomainAssociationResource.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.aws.ec2;
+
+import java.util.Set;
+
+import gyro.aws.AwsResource;
+import gyro.aws.Copyable;
+import gyro.core.GyroUI;
+import gyro.core.resource.Resource;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.ec2.model.TransitGatewayMulticastDomainAssociation;
+
+/**
+ * Creates a transit gateway multicast domain association.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *        association
+ *           vpc-attachment: $(aws::transit-gateway-vpc-attachment transit-gateway-vpc-attachment-example)
+ *           subnet: $(aws::subnet example-subnet-1)
+ *        end
+ */
+public class TransitGatewayMulticastDomainAssociationResource extends AwsResource
+    implements Copyable<TransitGatewayMulticastDomainAssociation> {
+
+    private TransitGatewayVpcAttachmentResource vpcAttachment;
+    private SubnetResource subnet;
+
+    /**
+     * The Vpc attachment that needs to be associated to the multicast domain. (Required)
+     */
+    @Required
+    public TransitGatewayVpcAttachmentResource getVpcAttachment() {
+        return vpcAttachment;
+    }
+
+    public void setVpcAttachment(TransitGatewayVpcAttachmentResource vpcAttachment) {
+        this.vpcAttachment = vpcAttachment;
+    }
+
+    /**
+     * The list of subnets that need to be included in the domain. (Required)
+     */
+    @Required
+    public SubnetResource getSubnet() {
+        return subnet;
+    }
+
+    public void setSubnet(SubnetResource subnet) {
+        this.subnet = subnet;
+    }
+
+    @Override
+    public void copyFrom(TransitGatewayMulticastDomainAssociation model) {
+        setVpcAttachment(findById(TransitGatewayVpcAttachmentResource.class, model.transitGatewayAttachmentId()));
+        setSubnet(findById(SubnetResource.class, model.subnet().subnetId()));
+    }
+
+    @Override
+    public String primaryKey() {
+        return String.format("%s in %s", subnet.getId(), vpcAttachment.getId());
```

Comment:
Use getters and setters for variables.
```suggestion
        return String.format("subnet %s in vpc attachment %s", getSubnet().getId(), getVpcAttachment().getId());
```